app/assets/javascripts/discourse/app/lib/flag-targets/flag.js

```diff
@@ -0,0 +1,57 @@
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class Flag {
+  flaggingTopic() {
+    return false;
+  }
+
+  isEditable() {
+    return true;
+  }
+
+  createFlag(controller, opts) {
```

Comment:
could probably just be `create`, it's little bit redundant to write `Flag.createFlag()`